auth/auth.go

```diff
@@ -0,0 +1,47 @@
+package auth
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/golang-jwt/jwt/v4"
+	"golang.org/x/oauth2"
+	goauth2 "google.golang.org/api/oauth2/v2"
+)
+
+type AuthService interface {
+	AuthenticateJWT(v url.Values) (AppToken, bool, error)
+	FetchUser(token *oauth2.Token) (*goauth2.Userinfo, error)
+}
+
+// An AppToken is the structure to make JWT authentication requests.
+type AppToken struct {
+	sc              jwt.RegisteredClaims
+	AccountAddress  string `json:"firmAddress"`
```

Comment:
these fields seem pretty implementation-specific to me. do they really belong in `trails`?